feat(Header): color token updates for Header

tulasigudibanda opened the pull request:
## Summary

Updated color tokens for Header

## ADO Story or GitHub Issue Link

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_queries/edit/2003564/?queryId=5e9196b5-2e3a-47b9-bb68-189413aca19d

## Figma Link

https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/branch/qMpff4GuFUEcsMUkvacS3U/Applications---Component-Library?node-id=8980-1081&m=dev


## To Do

![Screenshot 2025-01-23 at 2 52 41 PM](https://github.com/user-attachments/assets/ac418813-3126-4c20-ab07-ad0aed2efc1c)
![Screenshot 2025-01-23 at 2 53 50 PM](https://github.com/user-attachments/assets/acef1563-388b-41dd-ba68-29e381f367d9)
![image](https://github.com/user-attachments/assets/77738956-df74-47e6-9a87-929a721a9a66)

brian-patrick-3 reviewed (changes requested):
1. The colors for the nav menu, the various link states, and the flyout anchors all appear incorrect when comparing to figma.  
2. The flyout buttons all seem to have some extra whitespace below the icons. Might need to set display: block on those SVGs.

tulasigudibanda commented:
> 1. The colors for the nav menu, the various link states, and the flyout anchors all appear incorrect when comparing to figma.
> 2. The flyout buttons all seem to have some extra whitespace below the icons. Might need to set display: block on those SVGs.

@brian-patrick-3  , I had a call with Robert to check on these . He told me some changes to be done but looks like the figma is not updated.

brian-patrick-3 commented:
> @brian-patrick-3 , I had a call with Robert to check on these . He told me some changes to be done but looks like the figma is not updated.

Marco said he completed the global nav color changes this morning. I based on Local Nav changes on that if you'd like to compare. #322

tulasigudibanda commented:
> > @brian-patrick-3 , I had a call with Robert to check on these . He told me some changes to be done but looks like the figma is not updated.
> 
> Marco said he completed the global nav color changes this morning. I based my Local Nav changes on that if you'd like to compare. #322

ok. Will check and update .

tulasigudibanda commented:
@brian-patrick-3  , localNav background active token is --kd-color-background-menu-state-pressed . 
But for globalNav , figma shows background-button-primary-state-pressed . Which one should I go with?
![Screenshot 2025-01-23 at 4 09 22 PM](https://github.com/user-attachments/assets/b44a0d9f-da4f-48a2-8088-ff19380b7e04)

brian-patrick-3 commented:
I think you're looking at the wrong thing. You can find the link states in the other Global Nav section in Figma under Components

tulasigudibanda commented:
> I think you're looking at the wrong thing. You can find the link states in the other Global Nav section in Figma under Components

ok. It is actually split into mutiple places , with some getting updated and some are not : 
Components > Global Nav section 
Patterns >  Global Navigation Patterns (this one has header styles with logo etc)
Patterns > Notification Pattern
Components > Flyout Notification

brian-patrick-3 commented:
Focus on this section specifically: https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/branch/qMpff4GuFUEcsMUkvacS3U/Applications---Component-Library?node-id=8471-347420&m=dev

tulasigudibanda commented:
1.  Icon viewBox got reduced from viewBox="0 0 32 32" to viewBox="0 0 20 20" when carbon icon is changed to shidoka icon. I updated the 'top' style for the flyout menu.
2. Added margin on the icons to align with the text.
3. Added display:flex on the  .overflow-icon for center alignment.

tulasigudibanda commented:
> 1. The colors for the nav menu, the various link states, and the flyout anchors all appear incorrect when comparing to figma.
>    
>    1. The mobile back button in the nav links and flyouts should match the link styles.
> 2. The flyout buttons all seem to have some extra whitespace below the icons. Might need to set display: block on those SVGs.

For (2) , display: block  on svg does not work because ::slotted() cannot penetrate into the DOM structure of the slotted content. So I styled the wrapper(<span>)  instead with display:flex . Looks like carbon icons inherently have display:block on the svg elements .

brian-patrick-3 commented:
> For (2) , display: block on svg does not work because ::slotted() cannot penetrate into the DOM structure of the slotted content. So I styled the wrapper() instead with display:flex . Looks like carbon icons inherently have display:block on the svg elements .

Ah yes that is annoying. I do know a workaround for that, but it's kind of messy. After you import the SVG, it's in string format, so you can modify the string to inject the `slot` attribute onto the `<svg>` tag. That way the SVG is slotted directly without an extra `span` or something.

srpriyankashetty commented:
Hi @tulasigudibanda ,
Please check the padding between Mark All read and Setting icon in notification panel
![image](https://github.com/user-attachments/assets/2a17ada3-190c-4d2d-a9e1-dbe9cb445128)

srpriyankashetty commented:
Success text color in dark mode 
Success background color in light mode is not visible

![image](https://github.com/user-attachments/assets/770b0ef2-8d0a-4667-91fa-134b45e5ec0a)
![image](https://github.com/user-attachments/assets/78cccb44-2163-4a6c-80df-27df461898f7)

jaideep08 commented:
> Success text color in dark mode Success background color in light mode is not visible
> 
> ![image](https://private-user-images.githubusercontent.com/179849001/406956553-770b0ef2-8d0a-4667-91fa-134b45e5ec0a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc5ODgzMDIsIm5iZiI6MTczNzk4ODAwMiwicGF0aCI6Ii8xNzk4NDkwMDEvNDA2OTU2NTUzLTc3MGIwZWYyLThkMGEtNDY2Ny05MWZhLTEzNGI0NWU1ZWMwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyN1QxNDI2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZDM0NzFkNmRmYmMzNzE5YWE0NjdhOGYzNDFmZjI5NmVjOTlkYzBlOTcyMzI5MDMxY2ViYWExYzk4OTllOTM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3ayjrlly9RHxMZ4YfZZKWIeBzoCYNLH-w7QRe_dSiEg) ![image](https://private-user-images.githubusercontent.com/179849001/406956721-78cccb44-2163-4a6c-80df-27df461898f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc5ODgzMDIsIm5iZiI6MTczNzk4ODAwMiwicGF0aCI6Ii8xNzk4NDkwMDEvNDA2OTU2NzIxLTc4Y2NjYjQ0LTIxNjMtNGE2Yy04MGRmLTI3ZGY0NjE4OThmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyN1QxNDI2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZWNlOWE0MWQ3Mzg2YWI3Y2Y0NDVkZWY0ZTUxNzJjNmFiYjkwY2Y4ZGQ1NGQ4ZjZkMGNjNzE2OTg4YWMyZjE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KinU7b8bYmyd8vfy2APbeB7YLkzoVwmH9UotJfIgvyU)

The success text color issue is fixed

tulasigudibanda commented:
> between Mark All read and Setting icon in notification panel

Pushed the fix

brian-patrick-3 commented:
:tada: This PR is included in version 2.0.0-next.65 :tada:

The release is available on:
- [npm package (@next dist-tag)](https://www.npmjs.com/package/@kyndryl-design-system/shidoka-applications/v/2.0.0-next.65)
- [GitHub release](https://github.com/kyndryl-design-system/shidoka-applications/releases/tag/v2.0.0-next.65)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

brian-patrick-3 commented:
:tada: This PR is included in version 2.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@kyndryl-design-system/shidoka-applications/v/2.0.0)
- [GitHub release](https://github.com/kyndryl-design-system/shidoka-applications/releases/tag/v2.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
